Yafc/Workspace/ProductionTable/ProductionLinkSummaryScreen.cs

```diff
@@ -51,67 +51,70 @@ private void BuildScrollArea(ImGui gui) {
+                table = relationLinks[otherLink];
             }
-            var color = 1;
-            gui.spacing = 0.75f;
-            if (childLinks.Values.Any(e => e.Any())) {
-                gui.BuildText("Child links: ", Font.productionTableHeader);
-                foreach (var relTable in childLinks.Values) {
-                    BuildFlow(gui, relTable, relTable.Sum(e => Math.Abs(e.flow)), false, color++);
-                }
+            else {
+                table = unlinked;
             }
-            if (parentLinks.Values.Any(e => e.Any())) {
-                gui.BuildText("Parent links: ", Font.productionTableHeader);
-                foreach (var relTable in parentLinks.Values) {
-                    BuildFlow(gui, relTable, relTable.Sum(e => Math.Abs(e.flow)), false, color++);
-                }
+
+            table.Add((row, localFlow));
+        }
+        var color = 1;
+        gui.spacing = 0.75f;
+        if (childLinks.Values.Any(e => e.Any())) {
+            gui.BuildText("Child links: ", Font.productionTableHeader);
+            foreach (var relTable in childLinks.Values) {
+                BuildFlow(gui, relTable, relTable.Sum(e => Math.Abs(e.flow)), false, color++);
             }
-            if (otherLinks.Values.Any(e => e.Any())) {
-                gui.BuildText("Unrelated links: ", Font.productionTableHeader);
-                foreach (var relTable in otherLinks.Values) {
-                    BuildFlow(gui, relTable, relTable.Sum(e => Math.Abs(e.flow)), false, color++);
-                }
+        }
+        if (parentLinks.Values.Any(e => e.Any())) {
+            gui.BuildText("Parent links: ", Font.productionTableHeader);
+            foreach (var relTable in parentLinks.Values) {
+                BuildFlow(gui, relTable, relTable.Sum(e => Math.Abs(e.flow)), false, color++);
             }
-            if (unlinked.Any()) {
-                gui.BuildText("Unlinked: ", Font.subheader);
-                BuildFlow(gui, unlinked, 0, false);
+        }
+        if (otherLinks.Values.Any(e => e.Any())) {
+            gui.BuildText("Unrelated links: ", Font.productionTableHeader);
+            foreach (var relTable in otherLinks.Values) {
+                BuildFlow(gui, relTable, relTable.Sum(e => Math.Abs(e.flow)), false, color++);
+            }
+        }
+        if (unlinked.Any()) {
+            gui.BuildText("Unlinked: ", Font.subheader);
+            BuildFlow(gui, unlinked, 0, false);
+        }
+    }
+
+    public static ProductionTable? FindProductionTable(ModelObject link, out List<ModelObject> parents) {
+        ModelObject table = link;
+        parents = [];
+        while (table.ownerObject is not ProjectPage) {
+            var t = table.ownerObject;
+            if (t is null) {
+                break;
             }
+            table = t;
+            if (t == table.ownerObject) continue;
+            parents.Add(t);
         }
+        return table as ProductionTable;
```

Comment:
```suggestion
        return (ProductionTable)table;
```

Reply:
I think i prefer the null check here so things dont crash out if we ever fail to find the productionTable? Or should we just risk that under the assumption we'll always be able to find one?